frontend/src/components/Scratch/RenderTabs.tsx

```diff
@@ -0,0 +1,135 @@
+import { useRef } from "react"
+
+import { Compilation, Scratch } from "../../lib/api"
+import CompilerOpts from "../compiler/CompilerOpts"
+import Diff from "../Diff"
+import Editor from "../Editor"
+import { EditorInstance } from "../Editor/MonacoEditor"
+import ScoreBadge from "../ScoreBadge"
+import { Tab } from "../Tabs"
+
+import AboutScratch from "./AboutScratch"
+import styles from "./Scratch.module.scss"
+
+type ScratchTab = LeftScratchTab | RightScratchTab
+
+function renderTabs(tabs: {[i: number]: JSX.Element}, filter?: Array<ScratchTab>) {
+    if (!filter)
+        return Object.values(tabs)
+    else
+        return Object.keys(tabs).filter(key => filter.includes(+key)).map(key => tabs[key])
+}
+
+export enum LeftScratchTab {
+    SOURCE_CODE,
+    ABOUT,
+    CONTEXT,
+    SETTINGS,
+}
+
+export enum RightScratchTab {
+    DIFF,
+}
+
+/**
+ * Renders the left tabs of the scratch
+ * @param {Array<LeftScratchTab>} [filter=undefined] The tabs you want to render
+ * @returns Left tabs of scratch
+ */
+export function renderLeftTabs({ scratch, setScratch }: {
+    scratch: Scratch
+    setScratch: (s: Partial<Scratch>) => void
+}, filter?: Array<LeftScratchTab>): React.ReactElement<typeof Tab>[] {
+    const sourceEditor = useRef<EditorInstance>() // eslint-disable-line react-hooks/rules-of-hooks
+    const contextEditor = useRef<EditorInstance>() // eslint-disable-line react-hooks/rules-of-hooks
```

Comment:
We might be able to avoid these lint disables by naming this function `useLeftTabs` or something.